owner/src/main/java/org/the_chance/honeymart/ui/add_product/AddProductViewModel.kt

```diff
@@ -0,0 +1,115 @@
+import org.the_chance.honeymart.domain.util.ErrorHandler
+import org.the_chance.honeymart.domain.util.ValidationState
+import org.the_chance.honeymart.ui.base.BaseViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class AddProductViewModel @Inject constructor(
+    private val addProductUseCase: AddProductUseCase,
+    private val addProductImagesUseCase: AddProductImagesUseCase
+) : BaseViewModel<AddProductUiState, Unit>(AddProductUiState()), AddProductInteractionListener {
+    override val TAG: String = this::class.java.simpleName
+    private val categoryId = 43L
+
+    override fun addProduct(product: AddProductUiState) {
+        _state.update { it.copy(isLoading = true) }
+        tryToExecute(
+            {
+                addProductUseCase(
+                    product.name, product.price.toDouble(), product.description, categoryId,
+                ).toAddProductUiState()
+            },
+            onSuccess = ::onAddProductSuccess,
+            onError = ::onAddProductError
+        )
+    }
+
+    private fun onAddProductSuccess(product: AddProductUiState) {
+        _state.update {
+            it.copy(
+                error = null,
+                id = product.id,
+                name = product.name,
+                price = product.price,
+                description = product.description,
+            )
+        }
+        addProductImage(productId = product.id, images = _state.value.images)
+    }
+
+    private fun onAddProductError(errorHandler: ErrorHandler) {
+        _state.update { it.copy(isLoading = false) }
+        if (errorHandler is ErrorHandler.NoConnection) {
+            _state.update { it.copy(isLoading = false, isError = true) }
+        }
+    }
+
+    override fun addProductImage(productId: Long, images: List<ByteArray>) {
+        _state.update { it.copy(isLoading = true) }
+        tryToExecute(
+            { addProductImagesUseCase(productId, images) },
+            onSuccess = ::onAddProductImagesSuccess,
+            onError = ::onAddProductImagesError
+        )
+    }
+
+    private fun onAddProductImagesSuccess(message: String) {
+        _state.update { it.copy(isLoading = false, error = null) }
+    }
+
+    private fun onAddProductImagesError(errorHandler: ErrorHandler) {
+        _state.update { it.copy(isLoading = false) }
+        if (errorHandler is ErrorHandler.NoConnection) {
+            _state.update { it.copy(isLoading = false, isError = true) }
+        }
+    }
+
+    override fun onProductNameChanged(name: String) {
+        val productNameState: ValidationState = when {
+            name.isBlank() -> ValidationState.BLANK_TEXT_FIELD
+            name.length <= 2 -> ValidationState.SHORT_LENGTH_TEXT
+            else -> ValidationState.VALID_TEXT_FIELD
+        }
+        _state.update { it.copy(productNameState = productNameState, name = name) }
+    }
+
+    override fun onProductPriceChanged(price: String) {
```

Comment:
This function not single Responsible
handle check validation price and update price value need split to two functions.